app/invocation/invocation_build_logs_card.tsx

```diff
@@ -1,14 +1,137 @@
 import React from "react";
-import InvocationModel from "./invocation_model";
+import { eventlog } from "../../proto/eventlog_ts_proto";
+import { invocation } from "../../proto/invocation_ts_proto";
+import capabilities from "../capabilities/capabilities";
+import errorService from "../errors/error_service";
+import rpcService from "../service/rpc_service";
 import { TerminalComponent } from "../terminal/terminal";
+import { BuildBuddyError } from "../util/errors";
+import InvocationModel from "./invocation_model";
 
 interface Props {
   model: InvocationModel;
   expanded: boolean;
 }
 
-export default class BuildLogsCardComponent extends React.Component {
-  props: Props;
+interface State {
+  consoleBuffer?: string;
+  nextChunkId?: string;
+  fetchedFirstChunk?: boolean;
+}
+
+const POLL_TAIL_INTERVAL_MS = 1_000;
+const MAX_INITIAL_LINES = 100_000;
+
+const InvocationStatus = invocation.Invocation.InvocationStatus;
+
+export default class BuildLogsCardComponent extends React.Component<Props, State> {
+  state: State = {
+    consoleBuffer: "",
+    nextChunkId: "",
+    fetchedFirstChunk: false,
+  };
+
+  private pollTailTimeout: number | null = null;
+
+  constructor(props: Props) {
+    super(props);
+
+    const invocation = props.model.invocations[0];
+    if (!invocation) {
+      console.error("BuildLogsCard: invocation model is missing invocation");
+    }
+  }
+
+  componentDidMount() {
+    if (this.shouldFetchFromEventLog()) {
+      this.fetchTail(/*isFirstRequest=*/ true);
+    }
+  }
+
+  componentWillUnmount() {
+    window.clearTimeout(this.pollTailTimeout);
+  }
+
+  private shouldFetchFromEventLog() {
+    return capabilities.chunkedEventLogs && Boolean(this.props.model.invocations[0]?.lastChunkId);
+  }
+
+  private isInvocationComplete() {
```

Comment:
Seems like this method could live on model, that way we don't have to do this whole [0] thing.